numpy/lib/arraypad.py

```diff
@@ -1406,10 +1406,14 @@ def pad(array, pad_width, mode, **kwargs):
             newmat = _append_min(newmat, pad_after, chunk_after, axis)
 
     elif mode == 'reflect':
-        if narray.size == 0:
-            raise ValueError("There aren't any elements to reflect in `array`")
-
         for axis, (pad_before, pad_after) in enumerate(pad_width):
+            if newmat.shape[axis] == 0:
+                # Axes with non-zero padding cannot be empty.
+                if ((pad_before > 0) or (pad_after > 0)):
```

Comment:
I'd probably omit all the parens here. `if (...):` is just bad python that is trying to be C.

Reply:
Correct, fixed.